src/command.rs

```diff
@@ -1,36 +1,124 @@
-use clap::{AppSettings, Args, Parser, ValueEnum};
+// TODO: docstrings for everything!!!
 
-#[derive(Parser)]
-#[clap(name = "cargo")]
-#[clap(bin_name = "cargo")]
+use clap::{Args, Parser, Subcommand};
+
+#[derive(Parser, Debug)]
+#[command(name = "cargo")]
+#[command(bin_name = "cargo")]
 pub enum Cargo {
-    #[clap(name = "3ds")]
+    #[command(name = "3ds")]
     Input(Input),
 }
 
-#[derive(Args)]
-#[clap(about)]
-#[clap(global_setting(AppSettings::AllowLeadingHyphen))]
+#[derive(Args, Debug)]
+#[command(version, about)]
 pub struct Input {
-    #[clap(value_enum)]
+    /// The cargo command to run. This command will be forwarded to the real
+    /// `cargo` with the appropriate arguments for a 3DS executable.
+    #[command(subcommand)]
     pub cmd: CargoCommand,
-    pub cargo_opts: Vec<String>,
+
+    /// Don't actually run any commands, just echo them to the console.
+    /// This is mostly intended for testing.
+    #[arg(long, hide = true)]
+    pub dry_run: bool,
+
+    /// Pass additional options through to the `cargo` command.
+    ///
+    /// To pass flags that start with `-`, you must use `--` to separate `cargo 3ds`
+    /// options from cargo options. All args after `--` will be passed through
+    /// to cargo unmodified.
```

Comment:
This is a little unfortunate, but I guess it should work.

Reply:
Yeah, I'll keep exploring because the `clap` API is fairly extensive and there might be a way to handle it better with the right incantation of attributes.

I think ideally, I'd want:
* Any unknown args are forwarded to cargo, without requiring `--`
* If present, `--` separates cargo args from executable args

Does that make sense?